wrapanapi/systems/nuage.py

```diff
@@ -23,7 +23,8 @@ class NuageSystem(System):
         # We're returning 3rd element of .count() tuple which is formed as
         # entities.count() == (fetcher, served object, count of fetched objects)
         'num_security_group': lambda self: self.api.policy_groups.count()[2],
-        'num_cloud_subnet': lambda self: self.api.subnets.count()[2],
+        'num_cloud_subnet': lambda self: self.api.subnets.count(
+            filter="name != 'BackHaulSubnet'")[2] + self.api.l2_domains.count()[2],
```

Comment:
Consider leaving l2_domains as a separate stat, and in integration_tests add them together when comparing to the UI/API where they are combined?

Reply:
Thing is we reuse `validate_stats` function from as deep as `BaseProvider` ATM ([link](https://github.com/ManageIQ/integration_tests/blob/master/cfme/common/provider.py#L623-L653)) so it's a problem to customize it... That being said, if you think it makes sense, we can of course override the `validate_stats` function on NuageProvider itself and customize.

Reply:
This tugs at the root question of wrapanapi's usefulness beyond ManageIQ and integration_tests.

I've approved, but having an in-line comment here as to what's the reason behind excluding backhaul and then combining with l2_domain count will help us in the future when we fork wrapanapi for use beyond ManageIQ and move this logic into integration_tests and any other consumers of wrapanapi.

Reply:
Thanks for the quick review. I also added the comment explaining the reason for this change.